diff --git a/client/_utils.js b/client/_utils.js
index de4e3c4eedfaffa561ff7ea80f109cbf3bb807e5..acdcdb9a9b0bc1082c158cc4c74ef3684fd6ddf7 100644
--- a/client/_utils.js
+++ b/client/_utils.js
@@ -57,6 +57,7 @@ function _fetchData() {
               options.method = "POST";
             }
 
+            options.credentials = "include";
             _context.next = 7;
             return fetch(url, options);
 
@@ -81,7 +82,7 @@ function _fetchData() {
           case 16:
             _context.prev = 16;
             _context.t0 = _context["catch"](2);
-            logger.error("CLIENT_FETCH_ERROR", {
+            console.error("CLIENT_FETCH_ERROR", {
               error: _context.t0,
               url: url
             });
